api/cloud/oc_cloud_rd.c

```diff
@@ -233,13 +249,24 @@ delete_resources(oc_cloud_context_t *ctx)
   if (ctx->rd_delete_resources == NULL) {
     return;
   }
-
-  if (!rd_delete(ctx->cloud_ep, ctx->rd_delete_resources, ctx->device,
-                 delete_resources_handler, LOW_QOS, ctx)) {
-    OC_CLOUD_ERR("cannot send unpublish resource links request");
+  rd_links_partition_t partition;
+  memset(&partition, 0, sizeof(rd_links_partition_t));
+  if (rd_delete(ctx->rd_delete_resources, ctx->cloud_ep, ctx->device,
+                delete_resources_handler, LOW_QOS, ctx,
```

Comment:
Hm how you delete resources when you are not able delete all resources (partially). In `delete_resources_handler` there could be check when delete has been succeeded then you try to unpublish next resources.